circuit-lib/circuit-lib.circom/benchmarks/light/transaction_light.ts

```diff
@@ -0,0 +1,169 @@
+
+import { Keypair as SolanaKeypair, PublicKey } from "@solana/web3.js";
+import { BN } from "@coral-xyz/anchor";
+
+import {
+  Account,
+  Utxo,
+  FEE_ASSET,
+  Provider as LightProvider,
+  MINT,
+  Transaction,
+  TransactionParameters,
+  Action,
+  IDL_VERIFIER_PROGRAM_ZERO,
+  IDL_VERIFIER_PROGRAM_TWO,
+} from "@lightprotocol/zk.js";
+
+import { bs58 } from "@coral-xyz/anchor/dist/cjs/utils/bytes";
+import { MerkleTree } from "../../../circuit-lib.js/lib/index.js";
+import { stringifyBigInts } from "maci-crypto";
+
+// This will extract the type of the constructor parameters of Transaction CLass
+type TransactionInputs = ConstructorParameters<typeof Transaction>[0];
+
+function addPrivateKey(transaction: Transaction, account: Account) {
+  transaction.proofInput["inPrivateKey"] = transaction.params.inputUtxos.map(
+    (utxo: Utxo) => {
+      if (utxo.publicKey == account.pubkey) {
+        return account.privkey;
+      }
+    },
+  );
+}
+
+async function generateProofInputs(tree_height: number, app = false) {
+  const lightProvider = await LightProvider.loadMock();
+  const poseidon = await buildPoseidonOpt();
+
+  const seed32 = bs58.encode(new Uint8Array(32).fill(1));
+  const account = new Account({ poseidon: poseidon, seed: seed32 });
+  await account.getEddsaPublicKey();
+
+  const depositAmount = 20_000;
+  const depositFeeAmount = 10_000;
+
+  const deposit_utxo1 = new Utxo({
+    poseidon: poseidon,
+    assets: [FEE_ASSET, MINT],
+    amounts: [new BN(depositFeeAmount), new BN(depositAmount)],
+    publicKey: account.pubkey,
+    assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+    verifierProgramLookupTable:
+      lightProvider.lookUpTables.verifierProgramLookupTable,
+  });
+
+  const mockPubkey: PublicKey = SolanaKeypair.generate().publicKey;
+
+  let txParams = new TransactionParameters({
+    outputUtxos: [deposit_utxo1],
+    eventMerkleTreePubkey: mockPubkey,
+    transactionMerkleTreePubkey: mockPubkey,
+    senderSpl: mockPubkey,
+    senderSol: lightProvider.wallet.publicKey,
+    action: Action.SHIELD,
+    poseidon,
+    verifierIdl: app ? IDL_VERIFIER_PROGRAM_TWO : IDL_VERIFIER_PROGRAM_ZERO,
+    account,
+  });
+
+  lightProvider.solMerkleTree!.merkleTree = new MerkleTree(
+    tree_height,
+    poseidon,
+    [
+      deposit_utxo1.getCommitment(poseidon),
+      poseidon.F.toString(poseidon(["123124"])),
+    ],
+  );
+
+  let txInputs: TransactionInputs = {
+    ...(await lightProvider.getRootIndex()),
+    solMerkleTree: lightProvider.solMerkleTree!,
+    params: txParams,
+  };
+
+  if (app) {
+    txInputs.appParams = {
+      mock: "123",
+      verifierIdl: IDL_VERIFIER_PROGRAM_ZERO,
+    };
+  }
+
+  const tx = new Transaction(txInputs);
+  await tx.compile(lightProvider.poseidon, account);
+
+  addPrivateKey(tx, account);
+  if (app) delete tx.proofInput.inPublicKey;
+
+  return tx.proofInput;
+}
+
+async function mark_transaction_account(tree_height: number, app = false) {
```

Comment:
why mark...?
if you want to keep it short feel free to use bench...

Reply:
Additionally you could make this function a generic utility for any circuit benchmark by refactoring it to sth like this:
```
async function generateProof(
  name: string,
  generateInputsFunction: (...args: any[]) => Promise<any>,
  ...inputFunctionArgs: any[]
) {
  const wasm_path = `./artifacts/${name}/${name}.wasm`;
  const zkey_path = `./artifacts/${name}/${name}.zkey`;

  const proofInputs = await generateInputsFunction(...inputFunctionArgs);
  const res = await snarkjs.groth16.fullProve(
    stringifyBigInts(proofInputs),
    wasm_path,
    zkey_path
  );

  return res;
}
```

Reply:
mark is a notation to track a function's performance in general. I wrote following the benchmark framework as well. Anyway, I changed into `bench` for better readability.